code/game/machinery/cryopod.dm

```diff
@@ -97,290 +96,248 @@ var/global/list/frozen_items = list("Alpha"=list(),"Bravo"=list(),"Charlie"=list
-					data_core.security -= T
-					qdel(T)
-			for(var/datum/data/record/G in data_core.general)
-				if((G.fields["name"] == occupant.real_name))
-					data_core.general -= G
-					qdel(G)
-
-			if(orient_right)
-				icon_state = "body_scanner_0-r"
-			else
-				icon_state = "body_scanner_0"
-
-
-			occupant.ghostize(0) //We want to make sure they are not kicked to lobby.
-			//TODO: Check objectives/mode, update new targets if this mob is the target, spawn new antags?
-
-			//Make an announcement and log the person entering storage.
-			frozen_crew += "[occupant.real_name]"
-
-			announce.autosay("[occupant.real_name] has entered long-term hypersleep storage. Belongings moved to hypersleep inventory.", "Hypersleep Storage System")
-			visible_message("<span class='notice'>[src] hums and hisses as it moves [occupant.real_name] into hypersleep storage.</span>")
-
-			//Delete the mob.
-
-			qdel(occupant)
-			occupant = null
-			stop_processing()
+	if(occupant.stat == DEAD) //Occupant is dead, abort.
+		go_out()
+		return
 
+	//Allow a ten minute gap between entering the pod and actually despawning.
+	if(world.time - time_entered < time_till_despawn)
+		return
+
+	if(occupant.despawn(src))
+		QDEL_NULL(occupant)
+	stop_processing()
+	update_icon()
+
+/mob/proc/despawn(obj/machinery/cryopod/pod)
+	var/list/items = contents.Copy()
+
+	var/list/stored_items = list()
+
+	for(var/obj/item/W in items)
+		stored_items += W.place_in_storage()
+
+	var/list/dept_console = GLOB.cryoed_item_list["REQ"]
+	if(ishuman(src))
+		var/mob/living/carbon/human/H = src
+		switch(H.job)
+			if("Master at Arms","Command Master at Arms")
+				dept_console = GLOB.cryoed_item_list["MP"]
+			if("Medical Officer","Medical Researcher","Chief Medical Officer")
+				dept_console = GLOB.cryoed_item_list["Med"]
+			if("Ship Engineer","Chief Ship Engineer")
+				dept_console = GLOB.cryoed_item_list["Eng"]
+			else if(H.assigned_squad)
+				switch(H.assigned_squad.id)
+					if(ALPHA_SQUAD)
+						dept_console = GLOB.cryoed_item_list["Alpha"]
+					if(BRAVO_SQUAD)
+						dept_console = GLOB.cryoed_item_list["Bravo"]
+					if(CHARLIE_SQUAD)
+						dept_console = GLOB.cryoed_item_list["Charlie"]
+					if(DELTA_SQUAD)
+						dept_console = GLOB.cryoed_item_list["Delta"]
+
+	dept_console += stored_items
+
+	if(ishuman(src))
+		var/mob/living/carbon/human/H = src
+		if(H.mind && H.assigned_squad)
+			var/datum/squad/S = H.assigned_squad
+			switch(H.mind.assigned_role)
+				if("Squad Engineer")
+					S.num_engineers--
+				if("Squad Corpsman")
+					S.num_medics--
+				if("Squad Specialist")
+					S.num_specialists--
+					if(H.specset && !available_specialist_sets.Find(H.specset))
+						available_specialist_sets += H.specset //we make the set this specialist took if any available again
+
+				if("Squad Smartgunner")
+					S.num_smartgun--
+				if("Squad Leader")
+					S.num_leaders--
+			S.count--
+		H.assigned_squad?.clean_marine_from_squad(H,TRUE) //Remove from squad recods, if any.
```

Comment:
You can put that inside the `if(assigned_squad)` check once mind is out of the equation.